objects_and_classes/homework/homework.py

```diff
@@ -47,16 +47,230 @@
+        else:
+            self.cars = cars
+        self.places = int(places)
+        if owner is None:
+            self.owner = None
+        else:
+            self.owner = uuid.uuid4()
+        self.current = 0
+
+    def __contains__(self, item):
+        return item in self.cars
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+
+        if self.current < len(self.cars):
+            res = self.cars[self.current]
+            self.current += 1
+            return res
+        else:
+            self.current = 0
+            raise StopIteration
+
+    def __str__(self):
+        return f"Hello. I`m a garage in {self.town} and I have {self.places} cars inside me. They are: {self.cars}. " \
+            f"All of this is owned by {self.owner}."
+
+    def __repr__(self):
+        return "Garages('{}', '{}', {}, {})".format(self.town, self.cars, self.places, self.owner)
+
+    def __setstate__(self, state):
+        self.__dict__ = state
+
+    def __getstate__(self):
+        return self.__dict__
+
+    def get_len_cars(self):
+        return len(self.cars)
+
+    def get_cars(self):
+        return self.cars
+
+    def get_places(self):
+        return self.places
+
+    def add_car(self, car):
+        if len(self.cars) < self.places:
+            self.cars.append(car)
+        else:
+            print("No free space left!")
+            raise WrongException
+
+    def remove_car(self, car):
+        if car in self.cars:
+            self.cars.remove(car)
+
+    def hit_hat(self):
+        total = []
+        i = self.places
+        while i != 0:
+            for car in self.cars:
+                single = car.get_price()
+                total.append(single)
+                i -= 1
+            return sum(total)
+
+
+class Cesar:
+
+    def __init__(self, name, register_id, garages=None):
+        self.name = str(name)
+        self.register_id = uuid.uuid4()
+        if garages is None:
+            self.garages = []
+        else:
+            self.garages = garages
+
+    def __contains__(self, item):
+        return item in self.garages
+
+    def __str__(self):
+        return f"Hello. I`m a philanthropist, billionaire {self.name} and my ID is {self.register_id}. " \
+            f"I`m a proud owner of this cuties: {self.garages}."
+
+    def __repr__(self):
+        return "Cesar('{}', {}, '{}')".format(self.name, self.register_id, self.garages)
+
+    def __setstate__(self, state):
+        self.__dict__ = state
+
+    def __getstate__(self):
+        return self.__dict__
+
+    def total_hit_hat(self):
+        total = []
+        i = len(self.garages)
+        while i != 0:
+            for garage in self.garages:
+                single = garage.hit_hat()
+                total.append(single)
+                i -= 1
+            return sum(total)
+
+    def garage_count(self):
+        i = 0
+        j = len(self.garages)
+        while j != 0:
+            for garage in self.garages:
```

Comment:
looks so strange:) 
it is possible to do without a loop, using the length of the list

Reply:
Changed to boring "return len(self.garages)" )